### ISSUE TITLE
REST bridge: wrong MIME type for 400

### ISSUE THREAD

**Issue by oberstet** [crossbario/crossbar#785]
When using the **caller** part of the REST bridge, and when the called WAMP procedure produces an error, the bridge (correctly) returns a JSON thing with the WAMP error. It uses status code 400 (rather than 202 for a successful call). However, it does not set the MIME type as for 202: it is `text/html`.

Now, in retrospective, I'm not sure if it is a good idea to have a different status code (400) for WAMP app level errors anyway.

So either we change the behavior of the REST bridge to always return 202 (application/json) for any WAMP response and only return 400 on bad requests and 500 on internal errors OR the bridge should return the proper MIME type.

**Comment by oberstet** [comment]
As far as I see, we do use these HTTP response status codes: 200, 202, 400, 500

Maybe we can do with 3 instead?

* 200: WAMP processed (can either be a WAMP level success or a WAMP level error!)
* 400: bogus HTTP request (the client did wrong)
* 500: internal error (we screwed up .. there is a bug in CB)

**Comment by meejah** [comment]
Maybe not appropriate for this ticket, but related in a sense is a similar situation with the "transport" versus "wamp session" especially when connecting -- if we make a connection to crossbar successfully (at the transport level) but there's a WAMP-level error (e.g. "realm not found" when joining), we get an ``onLeave`` notification (i.e. WAMP level) of the error, but the transport closes cleanly.

So, to be "philosophically similar" in the ReST bridge, it would point to "always return 202" if the mechanics are correct (e.g. a valid HTTP request etc) but if there was a WAMP-layer error (e.g. not authorized or something) that comes out in the JSON --  and use 400/500 as described above.

The counter-point might be that it would make WAMP-layer errors more-obvious to clients if they were reflected in HTTP-level error-codes -- but that might be incorrectly blurring the line between transport (HTTP in this case) and WAMP...

TL;DR == I agree  your comment above makes ReST bridge more-consistent with rest of crossbar's behavior :)

**Comment by hawkowl** [comment]
This appears to be fixed in #799.
